src/Controls/src/Core/Layout/Layout.cs

```diff
@@ -96,23 +96,27 @@ public virtual void Add(IView child)
 			if (child == null)
 				return;
 
+			var index = _children.Count;
 			_children.Add(child);
 
 			if (child is Element element)
 			{
 				element.Parent = this;
-				VisualDiagnostics.OnChildAdded(this, element);
+				VisualDiagnostics.OnChildAdded(this, element, index);
 			}
 
-			AddToHandler(_children.Count, child);
+			AddToHandler(index, child);
```

Comment:
@hartez I am not sure which is correct. Previously, adding the first item would report that it added at index 1, which is incorrect I think?

Actually, all the adds would be one ahead, so the second view would be position 2 - or 3rd...

Reply:
Your version here is correct.